content/news/2020-11-13-release-plans-and-call-for-contrib.md

```diff
@@ -0,0 +1,48 @@
+type = "news"
+title = "R1 Status Update and Call for Contributors"
+author = "bitigchi"
+date = "2020-11-13 13:00:00+03:00"
+tags = ["r1"]
+
+After nearly 20 years of development, Haiku is close to its first stable release, namely R1. We need your help to make it happen. Here's a status update for R1 and the latest project news.
+
+## Release Plans
+
+We are planning to release Beta 3 sometime next year, hopefully followed by the stable release, R1. We currently have [31](https://dev.haiku-os.org/query?status=assigned&status=in-progress&status=reopened&status=new&group=status&milestone=R1%2Fbeta3) tickets left for Beta 3, and [688](https://dev.haiku-os.org/query?status=assigned&status=in-progress&status=reopened&status=new&group=status&milestone=R1) tickets waiting to be resolved. A big portion of these tickets are bug fixes that are aimed towards the general stability of the system.
+
+The Haiku project is by no means a small project, it is a full operating system after all. Maintaining a full-stack operating system is a huge amount of work, and needs more than a handful of developers. Recently, some of our core developers have announced that they will either limit their Haiku-time significantly due to other personal obligations or will take a break from the project. This jeopardizes the possibility of putting a stable release out in a viable timeframe.
```

Comment:
Again, there were no announcements. It would be better simply to cite the commit statistics, as it indeed appears I am not the only person to have scaled back.

Reply:
I agree, please remove this paragraph. People contribute to Haiku depending on their free time and motivation, and no one is essential. One or two developers leaving the project or scaling back does not jeopardize anything. You are putting pressure on people who don't need it, and the effect is very negative (you may have noticed that I have been mostly inactive for about two months, mainly because I don't like this climate and prefer to work on other things as long as Haiku feels like "oh no we need to ship a release, go back to work now!").

I appreciate your motivation and efforts to get things moving, but I perceive this kind of statement as an attack on the existing developers and quite counterproductive if your goal is to get (and keep) people involved.

Reply:
@pulkomandy, I am not putting pressure on anyone, this is about getting new people to get things moving. Like it or not, this is the current situation, and if you feel attacked, that's an issue on your side. Why would it bother you if you are only tinkering with Haiku on your free time, and not caring about a release (your words). If you don't like the climate, you can always turn off your notifications and work on Haiku as you please. Don't check the forums, don't check the IRC, don't check the mailing list. This is totally under your control. But you can't change what it really is, because these are things you'll hear people talking.

Again, this paragraph is not about you. But by reflecting your personal feelings here, you are inadvertently dramatising the subject. Please mind your business, and do not block people that are trying to move things forward.

@waddlesplash, is it okay for you if I just cite the commit numbers without providing actual stats, or you think it's necessary to provide some sort of actual number? I'd say the former would be okay, if not, I'd love some help with numbers.

Reply:
You use words like "jeopardizing" and "not viable" and I'm the one dramatizing.

Reply:
@bitigchi, are you are literally suggesting that if @pulkomandy does not like the climate you are creating (or, at least, trying to create) the solution is for him to ... leave the project?!

To be completely honest, I share many of Adrien's sentiments here. Your devotion to getting more contributors and otherwise improving Haiku is admirable; your push to get people to focus on what you find important or otherwise set deadlines, etc. is not, and indeed it is not helping. I have commented on this obliquely before, but if you are going to go so far as to literally suggest Adrien more or less leave the project, while saying things like "jeopardizes" or "viable timeframe", then I am going to make a much stronger statement than I have before:

Haiku is not your personal project, and you are not our boss. Either stop trying to boss people around, or we will be forced to sanction you. Again, your desire to see Haiku succeed on a grand scale is admirable, and we naturally share it. But the methods you are trying to use to get there are pretty problematic.

Reply:
This paragraph won't get us more developers. To me it looks like "we make plans for a release, knowing we don't have the people to do the work".

Reply:
@waddlesplash, I am not suggesting anyone to leave the project. Why would you get such a meaning when we are trying to get more contributors here? Do you care to explain? You can feel free to sanction me (like I care), but it's kind of interesting that you feel the need to be this defensive when you're not actively contributing to the project.

@pulkomandy, @waddlesplash, let's review the sentence in question:

> This jeopardizes the possibility of putting a stable release out in a viable timeframe.

I do not see what's the problem here. When the top two committer of the project "scales back" their time, it literally puts the project at risk, the risk of putting a release out in a "viable" timeframe. What's viable? 5 years? 10 years? I do not think so.

Reply:
> I am not suggesting anyone to leave the project.

You are suggesting that I stay away from the forums, IRC and mailing lists (and I guess you forgot comments on pul lrequests), which means I would not be participating in the project, but writing code on my own. So, yes, you did just that.
Haiku is not just about writing code and dropping it in the git repo, an important part of it is discussing what we're doing, sharing knowledge and ideas, deciding what to do, etc. Which is the part I value most (above shipping releases).

Also, Waddlesplash did say "more or less leave" and you are ignoring that nuance.

> When the top two committer of the project "scales back" their time, it literally puts the project at risk

It doesn't, as we have explained to you multiple times, the project is organized in such a way that people can come and go as they want (or, in fact, as they can), without putting the project at risk. It will just be less contributions from two developers, and more contributions from others (new or existing). Or maybe a few months of slower progress. People have been joining nad leaving Haiku for the last 20 years and I don't think it would be any different this time.

> What's viable? 5 years? 10 years? I do not think so.

That's bad news, because we are already 20 years in and still no release. We should have given up in 2006 then.

Reply:
Haiku has had periods of lots of activity and periods of less. Where other "hobbyist OS" projects have come and gone, we continue chugging along. I think we still have a _raison d'etre_, and insofar as there are people who share that, we will continue to progress, however slowly. So in that sense, if we continue at the pace we are at, we will eventually get somewhere, especially as consumer hardware "slows down" in terms of how it advances (it has been over a decade, and XHCI is still the USB controller standard, despite plenty of changes to USB since then, etc.) and it becomes much, much easier to "keep up".

It is very much worth noting, however, that it has never been part of Haiku's culture to "cut corners, get done quickly." We plod along, deliberating every major step of the way, pick what we think is the best solution, and then we implement it. That takes time, to be sure, but that is fundamentally what makes Haiku itself. That is why there are so many neat things, some of which are referenced in this very blog post, that we can be proud of: the package manager, the holistic system design, etc. etc. etc. Trying to pressure a cut-corners-and-get-done approach would be going against the very ethos that has gotten Haiku this far, and I think that is why @pulkomandy, @korli, myself, and many others have pushed so hard against it.

Again: Trying to attract new contributors is very good. But doing it for the wrong reasons or with the wrong intent is not good.